lib/pool/compression_cache_pool.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 openGemini Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"runtime"
+
+	"github.com/golang/snappy"
+	"github.com/klauspost/compress/zstd"
+)
+
+var (
+	gzipReaderPool = NewCachePool(nil, 2*runtime.NumCPU())
+
+	snappyReaderPool = NewCachePool(func() interface{} {
+		return snappy.NewReader(bytes.NewReader(nil))
+	}, 2*runtime.NumCPU())
+
+	zstdDecoderPool = NewCachePool(func() interface{} {
+		decoder, _ := zstd.NewReader(nil)
+		return decoder
+	}, 2*runtime.NumCPU())
+)
+
+func GetGzipReader(body []byte) (*gzip.Reader, error) {
```

Comment:
Define GET and PUT as interface or would it be better to use generics here?

Reply:
The put and get for each reader have different logic, I think outside is better. 
There is already defined put / get in the base.